osu.Game/Rulesets/Edit/ComposerDistanceSnapProvider.cs

```diff
@@ -195,22 +195,7 @@ public IEnumerable<TernaryButton> CreateTernaryButtons() => new[]
             new TernaryButton(DistanceSnapToggle, "Distance Snap", () => new SpriteIcon { Icon = OsuIcon.EditorDistanceSnap })
         };
 
-        protected override bool OnKeyDown(KeyDownEvent e)
-        {
-            if (e.Repeat)
-                return false;
-
-            handleToggleViaKey(e);
-            return base.OnKeyDown(e);
-        }
-
-        protected override void OnKeyUp(KeyUpEvent e)
-        {
-            handleToggleViaKey(e);
-            base.OnKeyUp(e);
-        }
-
-        private void handleToggleViaKey(KeyboardEvent key)
+        public void HandleToggleViaKey(KeyboardEvent key)
```

Comment:
What are these changes?

Reply:
I moved the event handling to `(Osu/Catch)HitObjectComposer` because the event for distance snap got handled before the select box handle could prevent the event from propagating. I assumed this was the best solution here because grid snap toggle was also being handled in `OsuHitObjectComposer` already.